src/payment/strategies/checkoutcom/checkoutcom-googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,128 @@
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import Payment from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategyActionCreator from '../../payment-strategy-action-creator';
+import { GooglePayPaymentProcessor, GooglePayPaymentStrategy } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomGooglepayPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    constructor(
+        _store: CheckoutStore,
+        _checkoutActionCreator: CheckoutActionCreator,
+        _paymentMethodActionCreator: PaymentMethodActionCreator,
+        _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        _paymentActionCreator: PaymentActionCreator,
+        _orderActionCreator: OrderActionCreator,
+        _googlePayPaymentProcessor: GooglePayPaymentProcessor
+    ) {
+        super(_store,
+            _checkoutActionCreator,
+            _paymentMethodActionCreator,
+            _paymentStrategyActionCreator,
+            _paymentActionCreator,
+            _orderActionCreator,
+            _googlePayPaymentProcessor);
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this.getGooglePayOptions()) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this.getStore().getState();
+
+        const testMode: boolean | undefined = getPaymentMethodOrThrow(methodId).config.testMode;
+        const checkoutcomKey: string = getPaymentMethodOrThrow(methodId).initializationData.checkoutcomkey;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this.getGooglePayOptions();
+
+        return Promise.resolve(this.getPayment())
+            .then(payment => {
```

Comment:
It seems there's too much going on in this method, too many nested `.then`
Might be better to split the functionality in private methods for a better understanding.

```js
if (!paymentData) {
  const paymentData  = await this._displayWallet();
}

const tokenizedPayment = await this._tokenizePayment(paymentData);

return this._submitOrder(tokenizedPayment);
```

Reply:
token conversion extracted as another method as suggested